thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/translator/ConfigTranslator.java

```diff
@@ -19,25 +19,30 @@
 
 package org.apache.pinot.thirdeye.detection.yaml.translator;
 
+import java.util.HashMap;
 import java.util.Map;
 import org.apache.pinot.thirdeye.datalayer.dto.AbstractDTO;
+import org.apache.pinot.thirdeye.detection.ConfigUtils;
 import org.apache.pinot.thirdeye.detection.DataProvider;
 import org.apache.pinot.thirdeye.detection.validators.ConfigValidator;
+import org.yaml.snakeyaml.Yaml;
 
 
 /**
  * Translate a raw yaml config and map it to a config object
  */
 public abstract class ConfigTranslator<T extends AbstractDTO, V extends ConfigValidator> {
 
-  Map<String, Object> yamlConfig;
-
   protected ConfigValidator validator;
   protected DataProvider dataProvider;
+  protected String yamlConfig;
+  protected Map<String, Object> yamlConfigMap;
+  protected Yaml yaml;
```

Comment:
could you make these instance variables final when possible?

Reply:
Done. I have also updated the interface making `translateConfig` accept the parsed config map as a parameter.